pylint/extensions/for_any_all.py

```diff
@@ -36,20 +41,53 @@ def visit_for(self, node: nodes.For) -> None:
             return
 
         if_children = list(node.body[0].get_children())
+        node_after_loop = node.next_sibling()
+
+        if self._assigned_reassigned_returned(node, if_children, node_after_loop):
+            final_return_bool = node_after_loop.value.name
+            suggested_string = self._build_suggested_string(node, final_return_bool)
+            self.add_message(
+                "consider-using-any-or-all",
+                node=node,
+                args=suggested_string,
+                confidence=HIGH,
+            )
+            return
         if not len(if_children) == 2:  # The If node has only a comparison and return
             return
         if not returns_bool(if_children[1]):
             return
 
         # Check for terminating boolean return right after the loop
-        node_after_loop = node.next_sibling()
+
         if returns_bool(node_after_loop):
             final_return_bool = node_after_loop.value.value
             suggested_string = self._build_suggested_string(node, final_return_bool)
             self.add_message(
-                "consider-using-any-or-all", node=node, args=suggested_string
+                "consider-using-any-or-all",
+                node=node,
+                args=suggested_string,
+                confidence=HIGH,
             )
 
+    @staticmethod
+    def _assigned_reassigned_returned(
+        node: nodes.For, if_children: list[nodes.NodeNG], node_after_loop: nodes.NodeNG
+    ) -> bool:
+        node_before_loop = node.previous_sibling()
+        if assigned_bool(node_before_loop):
```

Comment:
The indentation is pretty deep here, I would use premature return to simplify it.
```suggestion
        if not assigned_bool(node_before_loop):
            return False
```

Reply:
oh for sure, this is gnarly code just to get it to pass the tests.  Just wanted to see what you all thought before I tried to refactor.

I'll add changelog and see primer